dev/tasks/java-jars/github.yml

```diff
@@ -83,7 +83,7 @@ jobs:
           - { runs_on: ["macos-13"], arch: "x86_64"}
           - { runs_on: ["macos-14"], arch: "aarch_64" }
     env:
-      MACOSX_DEPLOYMENT_TARGET: "10.15"
+      MACOSX_DEPLOYMENT_TARGET: "14.0"
```

Comment:
That's quite the bump, I am pretty sure we also want to support less recent macos.
>Specify the minimum version of the target platform (e.g. macOS or iOS) on which the target binaries are to be deployed.

Reply:
Ah, my bad for merging this quickly. I was too eager to get java-jars working again. Let's revert that part of the diff since it's not necessary.

Reply:
> I was too eager to get java-jars working again.

No worry, it's a good goal^^ We can likely bump it to 11 or even 12  (as 11 is also eol... but we don't have a support policy so...)

Reply:
@assignUser this was an artifact from my previous failed attempt to fix this issue. I have not reverted that. But I will revert it shortly.